File: Computer Programming/C++/Hashing/Find All Duplicates in an Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) 
    {
        unordered_map<int,int>hash;
        for(auto it : nums)
        {
            hash[it] ++;
        }    
        vector<int> ans;
        for(auto it : nums)
        {
            if(hash[it] > 1)
            {
                ans.push_back(it);
                hash[it] = 0;
            }
        }
        return ans;
    }
};

File: Computer Programming/Python/Hashing/Find All Duplicates in an Array.py
class Solution(object):
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        
        mp = {}
        n = len(nums)

        for i in range(n):
            if nums[i] in mp:
                mp[nums[i]] += 1
            else:
                mp[nums[i]] = 1


        ans = set()

        for i in range(n):
            if mp[nums[i]] == 2:
                ans.add(nums[i])
    
        return list(ans)
